cluster-autoscaler/cloudprovider/aws/README.md

```diff
@@ -71,153 +92,158 @@ env:
 
-To run a CA pod in master node - CA deployment should tolerate the master `taint` and `nodeSelector` should be used to schedule the pods in master node.
-Please replace `{{ node_asg_min }}`, `{{ node_asg_max }}` and `{{ name }}` with your ASG setting in the yaml file.
 ```
-kubectl apply -f examples/cluster-autoscaler-run-on-master.yaml
+kubectl apply -f examples/cluster-autoscaler-autodiscover.yaml
 ```
 
-### Auto-Discovery Setup
+Cluster Autoscaler will respect the minimum and maximum values of each Auto
+Scaling Group. It will will only adjust the desired value.
 
-To run a cluster-autoscaler which auto-discovers ASGs with nodes use the `--node-group-auto-discovery` flag. For example, `--node-group-auto-discovery=asg:tag=k8s.io/cluster-autoscaler/enabled,k8s.io/cluster-autoscaler/<YOUR CLUSTER NAME>` will find the ASGs where those tag keys
-_exist_. It does not matter what value the tags have.
+Each Auto Scaling Group should be comprised of instance types that provide
+approximately equal capacity. For example, ASG "xlarge" could be comprised of
+m5a.xlarge, m4.xlarge, m5.xlarge, and m5d.xlarge instance types, because each of
+those provide 4 vCPUs and 16GiB RAM. Separately, ASG "2xlarge" could be
+comprised of m5a.2xlarge, m4.2xlarge, m5.2xlarge, and m5d.2xlarge instance
+types, because each of those provide 8 vCPUs and 32GiB RAM.
 
-Note that:
+Cluster Autoscaler will attempt to determine the CPU, memory, and GPU resources
+provided by an Auto Scaling Group based on the instance type specified in its
+Launch Configuration or Launch Template. It will also examine any overrides
+provided in an ASG's Mixed Instances Policy. If any such overrides are found,
+only the first instance type found will be used.  See [Using Mixed Instances
+Policies and Spot Instances](#Using-Mixed-Instances-Policies-and-Spot-Instances)
+for details.
 
-* It is recommended to use a second tag like `k8s.io/cluster-autoscaler/<YOUR CLUSTER NAME>` when `k8s.io/cluster-autoscaler/enabled` is used across many clusters to prevent ASGs from different clusters recognized as the node groups
-* There are no `--nodes` flags passed to cluster-autoscaler because the node groups are automatically discovered by tags
-* No min/max values are provided when using Auto-Discovery, cluster-autoscaler will respect the current min and max values of the ASG being targeted, and it will adjust only the "desired" value.
+From version 1.14, Cluster Autoscaler can also determine the resources provided
+by each Auto Scaling group via tags. The tag is of the format
+`k8s.io/cluster-autoscaler/node-template/resources/<resource-name>`.
+`<resource-name>` is the name of the resource, such as `ephemeral-storage`. The
+value of each tag specifies the amount of resource provided. The units are
+identical to the units used in the `resources` field of a Pod specification.
 
-```
-kubectl apply -f examples/cluster-autoscaler-autodiscover.yaml
-```
+Example tags:
 
-## Scaling a node group to 0
```

Comment:
This is only being used by `scale from 0` case. Why do you delete header?

Reply:
The previous language suggested that if there were no pods scheduled on an ASG that there might be a reason for the ASG size not to scale down to zero, and that it could be corrected somehow by tagging.  Or, that scaling up from zero was an important special case. Having re-read the documentation and the source code a number of times, I'm not sure that scaling down to zero or up from zero is a special case. Am I mistaken?